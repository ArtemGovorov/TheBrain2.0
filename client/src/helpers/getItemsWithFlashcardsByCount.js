export default function getItemsWithFlashcardsByCount (flashcards) {
  const itemsWithFlashcardsByCount = {
    newDone: 0,
    newTotal: 0,
    dueDone: 0,
    dueTotal: 0,
    reviewDone: 0,
    reviewTotal: 0
  }

  flashcards.forEach(({ item }) => {
    if (item.actualTimesRepeated === 1) {
      itemsWithFlashcardsByCount['newDone']++
    }
    if (item.extraRepeatToday) {
      itemsWithFlashcardsByCount['reviewTotal']++
    } else if (item.timesRepeated === 1) {
      itemsWithFlashcardsByCount['newDone']++
      if (item.actualTimesRepeated > 1) {
        itemsWithFlashcardsByCount['reviewDone']++
      }
    }
    if (item.nextRepetition) {
      if (item.actualTimesRepeated > 1) {
        itemsWithFlashcardsByCount['dueDone']++
      }
      itemsWithFlashcardsByCount['dueTotal']++
    } else {
      itemsWithFlashcardsByCount['newTotal']++
    }
  })

  return itemsWithFlashcardsByCount
}
