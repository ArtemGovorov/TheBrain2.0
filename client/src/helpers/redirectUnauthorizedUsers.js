import { push } from 'react-router-redux'

export default function redirectUnauthorizedUsers(props) {
  if (props.currentItems.loading || props.currentUser.loading) {
    return
  }

  const itemsWithFlashcard = props.currentItems.ItemsWithFlashcard

  if (itemsWithFlashcard.length > 0) {
    return
  }

  if (props.currentUser.activated) {
    props.dispatch(push('/'))
  } else {
    props.dispatch(push('/signup'))
  }
}
