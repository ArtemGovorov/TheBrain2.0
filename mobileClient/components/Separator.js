import React from 'react'
import { View } from 'react-native'

const Separator = (props) => (
  <View style={[style.line, { height: props.height, backgroundColor: props.color }]}>
    <View style={[style.circle, getSize(props), { backgroundColor: props.color }]} />
    <View style={[style.circle, getSize(props), { right: 0, backgroundColor: props.color }]} />
  </View>
)

Separator.defaultProps = {
  color: '#999',
  height: 1,
  circleSize: 10
}

const style = {
  line: {
    position: 'relative',
    zIndex: 1000
  },
  circle: {
    position: 'absolute'
  }
}

function getSize ({ circleSize, height }) {
  return {
    top: -circleSize / 2 + height / 2,
    width: circleSize,
    height: circleSize,
    borderRadius: circleSize
  }
}

export default Separator
