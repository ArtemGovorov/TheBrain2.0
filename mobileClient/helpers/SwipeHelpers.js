// @flow

export function calculateSwipeDirection(x: number, y: number) {
  const angleDeg = Math.atan2(y - 0, x - 0) * 180 / Math.PI;
  return (Math.round(angleDeg / 90) + 2) % 4 + 1;
}

export function calculateDragLength (x: number, y: number) {
  return Math.sqrt((Math.pow(y, 2) + Math.pow(x, 2)));
}
