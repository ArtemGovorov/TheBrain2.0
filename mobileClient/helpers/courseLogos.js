export default {
  'Chemistry': { file: require(`../images/chemistry.svg`), scale: 1 },
  'Biology': { file: require(`../images/biology.svg`), scale: 0.75 },
  'Computer Science': { file: require(`../images/chemistry.svg`), scale: 1 },
  'History': { file: require(`../images/biology.svg`), scale: 0.75 }
}
