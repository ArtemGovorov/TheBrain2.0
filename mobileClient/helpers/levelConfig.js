export default {
  0: { file: require('../images/TB_level1.png'), name: 'infant' },
  1: { file: require('../images/TB_level1.png'), name: 'baby' },
  2: { file: require('../images/TB_level2.png'), name: 'youngster' },
  3: { file: require('../images/TB_level3.png'), name: 'child' },
  4: { file: require('../images/TB_level4.png'), name: 'teen' },
  5: { file: require('../images/TB_level5.png'), name: 'student' },
  6: { file: require('../images/TB_level6.png'), name: 'grownup' },
  7: { file: require('../images/TB_level7.png'), name: 'specialist' },
  8: { file: require('../images/TB_level8.png'), name: 'expert' }
}
