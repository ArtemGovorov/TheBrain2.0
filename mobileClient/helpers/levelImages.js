export default {
  1: { file: require('../images/TB_level1.png') },
  2: { file: require('../images/TB_level2.png') },
  3: { file: require('../images/TB_level3.png') },
  4: { file: require('../images/TB_level4.png') },
  5: { file: require('../images/TB_level5.png') },
  6: { file: require('../images/TB_level6.png') },
  7: { file: require('../images/TB_level7.png') },
  8: { file: require('../images/TB_level8.png') }
}
