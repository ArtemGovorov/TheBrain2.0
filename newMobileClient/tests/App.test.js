import Separator from '../components/Separator'

require('./helpers/setup')
const React = require('react')

// jest.mock('react-native', () => require('react-native-mock-render'), {virtual: true})
// const renderer = require('react-native-mock-render/mock');
const Enzyme = require('enzyme');

it('renders without crashing', () => {
  const Seperator = require('../components/Separator.js')
  const wrapper = Enzyme.mount(<Separator/>)

  expect(wrapper.html()).toMatchSnapshot()
  // const rendered = renderer.create(<App />).toJSON();
  // expect(rendered).toBeTruthy();
});
