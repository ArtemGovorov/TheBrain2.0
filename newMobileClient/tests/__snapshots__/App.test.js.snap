// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `"<view style=\\"height: 10px; width: 90%;\\"><view></view><view></view><view></view></view>"`;
